src/services/refactors/convertStringOrTemplateLiteral.ts

```diff
@@ -147,47 +147,66 @@ namespace ts.refactor.convertStringOrTemplateLiteral {
         }
     };
 
-    function concatConsecutiveString(index: number, nodes: readonly Expression[]): [number, string, number[]] {
+    function escapeStringForTemplate(s: string) {
+        // Escaping for $s in strings that are to be used in template strings
+        // Naive implementation: replace \x by itself and otherwise $ by \$.
+        // But to complicate it a bit, this should work for raw strings too.
+        // And another bit: escape the $ in the replacement for JS's .replace().
+        return s.replace(/\\.|\$/g, m => m === "$" ? "\\\$" : m);
+        // Finally, a less-backslash-happy version can work too, doing only ${ instead of all $s:
+        //     s.replace(/\\.|\${/g, m => m === "${" ? "\\\${" : m);
+        // but `\$${foo}` is likely more clear than the more-confusing-but-still-working `$${foo}`.
+    }
+
+    function getRawTextOfTemplate(node: TemplateHead | TemplateMiddle | TemplateTail) {
+        // in these cases the right side is ${
+        const rightShaving = isTemplateHead(node) || isTemplateMiddle(node) ? -2 : -1;
+        return getTextOfNode(node).slice(1, rightShaving);
+    }
+
+    function concatConsecutiveString(index: number, nodes: readonly Expression[]): [number, string, string, number[]] {

```

Comment:
Just a suggestion for readability on the consumer side:

```suggestion
    function concatConsecutiveString(index: number, nodes: readonly Expression[]): [nextIndex: number, text: string, rawText: string, usedIndexes: number[]] {
```